airflow/models/pool.py

```diff
@@ -114,7 +114,7 @@ def create_or_update_pool(
         if not name:
             raise ValueError("Pool name must not be empty")
 
-        pool = session.query(Pool).filter_by(pool=name).one_or_none()
+        pool = session.scalars(select(Pool).filter_by(pool=name)).one_or_none()
```

Comment:
```suggestion
        pool = session.scalar(select(Pool).filter_by(pool=name))
```

for consistency (this is the style we’re using elsewhere)